newa/cli.py

```diff
@@ -624,6 +626,30 @@ def worker(ctx: CLIContext, schedule_file: Path) -> None:
     ctx.save_execute_job('execute-', execute_job)
 
 
+def wait_for_requests_to_finish(ctx: CLIContext, requests: list[TFRequest]) -> None:
+    delay = int(ctx.settings.tf_recheck_delay)
+    watchlist = requests.copy()
+    while watchlist:
+        next_watchlist = []
```

Comment:
I don't think so. Changing the content of a list while iterating over it is not recommended, you may get lucky & things will work, and next Tuesday it might erase your hard drive... Using a "future list" for collecting items and swapping it for `while` to test it looks very appropriate.